p2p/transport/webrtc/errors.go

```diff
@@ -0,0 +1,49 @@
+package libp2pwebrtc
+
+import (
+	"fmt"
+)
+
+type errKind string
+
+const (
+	errKindConnectionFailed errKind = "peerconnection failed"
+	errKindDatachannel      errKind = "datachannel"
+	errKindMultiaddr        errKind = "bad multiaddr"
+	errKindNoise            errKind = "noise"
+	errKindInternal         errKind = "internal"
+)
+
+var (
+	errDataChannelTimeout = errDatachannel("timed out waiting for datachannel", nil)
+)
+
+type webRTCTransportError struct {
```

Comment:
We are using similar structured errors in the js implementation. I feel it also neatly categorizes the kinds of errors we encounter in this transport implementation. It will become abstracted by the `error` type anyway.

Reply:
Got it. Thank you!

Reply:
@Jorropo not sure how to surface the nested error using `fmt.Errorf`. Do I just do `fmt.Errorf("peerconnection failed: %s %v", msg, err)` for nested error?